File: apps/dotnet/Functions/BlobProcessorFunction.cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Blob upload processor — triggered when a blob is uploaded to the uploads container.
/// Binding: Blob trigger on "uploads/{name}" with AzureWebJobsStorage connection.
/// </summary>
public class BlobProcessorFunction
{
    private readonly ILogger<BlobProcessorFunction> _logger;

    public BlobProcessorFunction(ILogger<BlobProcessorFunction> logger)
    {
        _logger = logger;
    }

    [Function("blobProcessor")]
    public void Run(
        [BlobTrigger("uploads/{name}", Connection = "AzureWebJobsStorage")] string content,
        string name)
    {
        _logger.LogInformation("Blob processed: {Name}, size: {Size} chars", name, content.Length);
    }
}

File: apps/java/src/main/java/com/functions/BlobProcessorFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;

/**
 * Blob trigger — processes blobs uploaded to the 'uploads' container.
 */
public class BlobProcessorFunction {

    @FunctionName("blobProcessor")
    @StorageAccount("AzureWebJobsStorage")
    public void run(
            @BlobTrigger(
                name = "content",
                path = "uploads/{name}",
                connection = "AzureWebJobsStorage",
                source = "EventGrid")
            byte[] content,
            @BindingName("name") String name,
            final ExecutionContext context) {

        context.getLogger().info("Processing blob: " + name + ", size: " + content.length + " bytes");
    }
}
